internal/core/command/controller/messaging/external/external_test.go

```diff
@@ -206,8 +217,122 @@ func Test_commandQueryHandler(t *testing.T) {
 	}
 }
 
-func testPayload() types.MessageEnvelope {
+func Test_commandRequestHandler(t *testing.T) {
+	deviceResponse := responses.DeviceResponse{
+		BaseResponse: commonDTO.NewBaseResponse("", "", http.StatusOK),
+		Device: dtos.Device{
+			Name:        testDeviceName,
+			ProfileName: testProfileName,
+			ServiceName: testDeviceServiceName,
+		},
+	}
+	deviceServiceResponse := responses.DeviceServiceResponse{
+		BaseResponse: commonDTO.NewBaseResponse("", "", http.StatusOK),
+		Service: dtos.DeviceService{
+			Name: testDeviceServiceName,
+		},
+	}
+
+	lc := &lcMocks.LoggingClient{}
+	lc.On("Error", mock.Anything).Return(nil)
+	lc.On("Debugf", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	dc := &clientMocks.DeviceClient{}
+	dc.On("DeviceByName", context.Background(), testDeviceName).Return(deviceResponse, nil)
+	dsc := &clientMocks.DeviceServiceClient{}
+	dsc.On("DeviceServiceByName", context.Background(), testDeviceServiceName).Return(deviceServiceResponse, nil)
+	client := &mocks.MessageClient{}
+	client.On("Publish", mock.Anything, mock.Anything).Return(nil)
+	dic := di.NewContainer(di.ServiceConstructorMap{
+		container.ConfigurationName: func(get di.Get) interface{} {
+			return &config.ConfigurationStruct{
+				Service: bootstrapConfig.ServiceInfo{
+					Host:           mockHost,
+					Port:           mockPort,
+					MaxResultCount: 20,
+				},
+				MessageQueue: config.MessageQueue{
+					Required: true,
+					Internal: bootstrapConfig.MessageBusInfo{
+						Topics: map[string]string{
+							RequestTopicPrefix: "unittest/internal/request/",
```

Comment:
OK, you are assuming the prefix has the ending `/`.
This has not be the practice . See here:
https://github.com/edgexfoundry/device-virtual-go/blob/main/cmd/res/configuration.toml#L56
We need to stay consistent with not putting the  ending `/` on the prefix.

Reply:
Glad to hear it ! ( was actually following the ADR example)